packages/databricks-vscode/src/configuration/types.ts

```diff
@@ -43,7 +58,8 @@ export function isBundleConfigKey(key: any): key is keyof BundleFileConfig {
     return BUNDLE_FILE_CONFIG_KEYS.includes(key);
 }
 
-export interface ConfigReaderWriter<T extends keyof DatabricksConfig> {
-    readAll(target: string): Promise<DatabricksConfig | undefined>;
-    write(key: T, target: string, value?: DatabricksConfig[T]): Promise<void>;
+export function isAuthenticatedBundleConfigKey(
+    key: any
+): key is keyof AuthenticatedBundleConfig {
```

Comment:
But the AuthenticatedBundleConfig includes the keys from the DatabricksConfig, so this condition is less strict that our actual logic below. Is that intentional?

Reply:
Yes. The idea is to be able to quickly disambiguate if a key belongs to one of the models, and can hence be safely passed around.